src/test/java/io/jenkins/plugins/analysis/core/history/PreviousRunReferenceTest.java

```diff
@@ -282,30 +269,25 @@ void twoTrueAction_withAnyResult_shouldReturnTwo() {
         PreviousRunReference previousRun = new PreviousRunReference(baseline, selector,
                 false);
 
-        Optional<ResultAction> resultAction1 = previousRun.getReferenceAction();
-        Optional<ResultAction> resultAction2 = previousRun.getReferenceAction();
-
+        assertThat(previousRun.getReferenceAction()).isEqualTo(Optional.of(action));
+        assertThat(previousRun.getReferenceAction()).isEqualTo(Optional.of(action));
         verify(selector, times(4)).get(previousRunMock);
```

Comment:
this test, just tests two times the same way through the code, so you can simplify it by replacing

assertThat(previousRun.getReferenceAction()).isEqualTo(Optional.of(action));
assertThat(previousRun.getReferenceAction()).isEqualTo(Optional.of(action));
verify(selector, times(4)).get(previousRunMock);

by

assertThat(previousRun.getReferenceAction()).contains(action);
verify(selector, times(2)).get(previousRunMock);

Since you actually just test one resultAction you should change the title and description, too.